base/combinatorics.jl

```diff
@@ -125,7 +125,7 @@ function invpermute!!(a, p::AbstractVector{<:Integer})
     count = 0
     start = 0
     while count < length(a)
-        start = findnext(!iszero, p, start+1)
+        start = findnext(!iszero, p, start+1)::Int
```

Comment:
Are we actually sure all `AbstractArray` implementations will return an `Int`? You could also try using `notnothing(...)`, which throws when `nothing` is passed to it, which should allow the compiler to assert the type (need to check that).

BTW, how did you find the places where this assertion can be added?

Reply:
Anyplace where the code didn't seem to check for a 0 return but then indexed it, I assumed it was guaranteed to succeed.